wvdecrypter/cdm/media/cdm/cdm_adapter.cc

```diff
@@ -70,19 +70,17 @@ void* GetCdmHost(int host_interface_version, void* user_data)
 
 }  // namespace
 
-std::atomic<bool> exit_thread_flag;
-std::atomic<bool> timer_thread_running;
-
-void timerfunc(std::shared_ptr<CdmAdapter> adp, uint64_t delay, void* context)
+void CdmAdapter::timerfunc(CdmAdapter* adp, int64_t delay, void* context)
 {
-  timer_thread_running  = true;
-  uint64_t waited = 0;
-  std::this_thread::sleep_for(std::chrono::milliseconds(delay));
-  if (!exit_thread_flag) 
   {
-    adp->TimerExpired(context);
+    std::unique_lock<std::mutex> lock(m_closeSessionMutex);
+    auto predicate = [this] { return ms_isCloseSession.load(); };
+    const bool isClosing =
+      ms_sessionClosingCond.wait_for(lock, std::chrono::milliseconds(delay), predicate);
```

Comment:
predicate can stay inline
```suggestion
    const bool isClosing = ms_sessionClosingCond.wait_for(
        lock, std::chrono::milliseconds(delay), [this] { return ms_isCloseSession.load(); });
```